libpromises/generic_agent.c

```diff
@@ -762,16 +762,10 @@ void GenericAgentInitialize(EvalContext *ctx, GenericAgentConfig *config)
 
     if (config->agent_specific.agent.bootstrap_policy_server)
     {
-        snprintf(vbuff, CF_BUFSIZE, "%s%cfailsafe.cf", GetInputDir(), FILE_SEPARATOR);
-
-        if (stat(vbuff, &statbuf) == -1)
+        if (CheckAndGenerateFailsafe(GetInputDir(), "failsafe.cf"))
         {
             GenericAgentConfigSetInputFile(config, GetInputDir(), "failsafe.cf");
         }
-        else
```

Comment:
Actually the previous behavior was wrong. If failsafe.cf doesn't exist (stat returns -1) we are setting failsafe.cf to be exactly the same file which was checked to be not there.
What is more, in GenericAgentDiscoverContext() in case of bootstrap we are wiping out whole inputs directory (RemoveAllExistingPolicyInInputs()) and then we are generating failsafe.cf (GenericAgentDiscoverContext()). So I will right now set failsafe.cf to be policy input file there. 
Thanks for noticing this one.